packages/nouns-contracts/contracts/NounsAuctionHouseV2.sol

```diff
@@ -0,0 +1,511 @@
+     */
+    function warmUpSettlementState(uint256[] calldata nounIds) external {
+        for (uint256 i = 0; i < nounIds.length; ++i) {
+            if (settlementHistory[nounIds[i]].blockTimestamp == 0) {
+                settlementHistory[nounIds[i]] = SettlementState({ blockTimestamp: 1, amount: 0, winner: address(0) });
+            }
+        }
+    }
+
+    /**
+     * @notice Get past auction settlements.
+     * @dev Returns settlements in reverse order, meaning settlements[0] will be the most recent auction price.
+     * Skips auctions where there was no winner, i.e. no bids.
+     * @param auctionCount The number of price observations to get.
+     * @return settlements An array of type `Settlement`, where each Settlement includes a timestamp,
+     * the Noun ID of that auction, the winning bid amount, and the winner's address.
+     */
+    function getSettlements(uint256 auctionCount) external view returns (Settlement[] memory settlements) {
+        uint256 latestNounId = auctionStorage.nounId;
+        if (!auctionStorage.settled && latestNounId > 0) {
+            latestNounId -= 1;
+        }
+
+        settlements = new Settlement[](auctionCount);
+        uint256 actualCount = 0;
+        while (actualCount < auctionCount && latestNounId > 0) {
+            SettlementState memory settlementState = settlementHistory[latestNounId];
+            // Skip Nouner reward Nouns, they have no price
+            // Also skips IDs with no price data
+            if (settlementState.winner == address(0)) {
+                --latestNounId;
+                continue;
+            }
+
+            settlements[actualCount] = Settlement({
+                blockTimestamp: settlementState.blockTimestamp,
+                amount: uint64PriceToUint256(settlementState.amount),
+                winner: settlementState.winner,
+                nounId: latestNounId
+            });
+            ++actualCount;
+            --latestNounId;
+        }
+
+        if (auctionCount > actualCount) {
+            // this assembly trims the observations array, getting rid of unused cells
+            assembly {
+                mstore(settlements, actualCount)
+            }
+        }
+    }
+
+    /**
+     * @notice Get past auction prices.
+     * @dev Returns prices in reverse order, meaning prices[0] will be the most recent auction price.
+     * Skips auctions where there was no winner, i.e. no bids.
+     * @param auctionCount The number of price observations to get.
+     * @return prices An array of uint256 prices.
+     */
+    function getPrices(uint256 auctionCount) external view returns (uint256[] memory prices) {
+        uint256 latestNounId = auctionStorage.nounId;
+        if (!auctionStorage.settled && latestNounId > 0) {
+            latestNounId -= 1;
+        }
+
+        prices = new uint256[](auctionCount);
+        uint256 actualCount = 0;
+        while (actualCount < auctionCount && latestNounId > 0) {
```

Comment:
May be worth optimizing this loop as it could run 90+ times in the proposed design.

e.g.

```solidity
SettlementState memory settlementState;
uint256 actualCount;
for (uint256 id = latestNounId; id > 0 && actualCount < auctionCount; --id) {
    if (id % 10 == 0 && id <= 1820) continue; // Skip Nounder nouns.

    settlementState = settlementHistory[id];
    if (settlementState.winner == address(0)) continue; // Skip IDs with no price data.

    prices[actualCount] = uint64PriceToUint256(settlementState.amount);
    unchecked {
        ++actualCount;
    }
}
```